llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -8456,6 +8458,26 @@ LegalizerHelper::lowerAbsToCNeg(MachineInstr &MI) {
   return Legalized;
 }
 
+LegalizerHelper::LegalizeResult LegalizerHelper::lowerFAbs(MachineInstr &MI) {
+  Register SrcReg = MI.getOperand(1).getReg();
+  Register DstReg = MI.getOperand(0).getReg();
+
+  LLT Ty = MRI.getType(DstReg);
+  if (MRI.getType(SrcReg) != Ty)
+    return UnableToLegalize;
```

Comment:
Remove this check, the IR is malformed if this happens

Reply:
Removed